instrumentation/nocode/src/main/java/com/splunk/opentelemetry/instrumentation/nocode/NocodeAttributesExtractor.java

```diff
@@ -41,9 +41,18 @@ public void onStart(
     Map<String, String> attributes = mi.getRuleAttributes();
     for (String key : attributes.keySet()) {
       String expression = attributes.get(key);
-      String value = NocodeEvaluation.evaluate(expression, mi.getThiz(), mi.getParameters());
-      if (value != null) {
-        attributesBuilder.put(key, value);
+      Object value = NocodeEvaluation.evaluate(expression, mi.getThiz(), mi.getParameters());
+      if (value instanceof Long
```

Comment:
Something like this has come up in a couple of places in upstream repository. Once you have the primitives and their wrappers covered the question is why stop there, otel also has attributes with array types. For example https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/fdcf0d264eb0a9b2e3a79cf01753e2869aafd295/instrumentation/logback/logback-appender-1.0/library/src/main/java/io/opentelemetry/instrumentation/logback/appender/v1_0/internal/LoggingEventMapper.java#L295 handles a bunch of array types. If you choose not to handle these you could add a comment.